roscala/src/main/scala/coop/rchain/rosette/prim/tuple.scala

```diff
@@ -63,6 +63,31 @@ object tuple {
     }
   }
 
+  /**
+    * Define the tuple-concat primitive.
+    * This concatenates n Tuples
+    * e.g. (tuple-concat [1 2 3] [4] [5 6]) ==> [1 2 3 4 5 6]
+    */
+  object tplConcat extends Prim {
+    override val name: String = "tuple-concat"
+    override val minArgs: Int = 0
+    override val maxArgs: Int = MaxArgs
+
+    @checkTypeMismatch[Tuple] // All args must be Tuples
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
+      val elem = ctxt.argvec.elem
+      val n = ctxt.nargs
+      var ret = Tuple(Seq.empty)
```

Comment:
We try to avoid the usage of `var`. Can you do that with `fold`?

Reply:
Maybe when I'm better at Scala.  I tried for a while, and couldn't make it work.

Reply:
@tymm I've converted this to use foldLeft.  Can you PTAL?